### DISCUSSION TITLE
Dancing wave recording

### DISCUSSION THREAD

**Discussion by fatehux** [katspaugh/wavesurfer.js#3625] [Q&A]
Hello bro,
When i use record plugin, wave recording go to dancing up and down and not fixed, it is seems vibrating wave how can i prevent that?

**Comment by katspaugh** [comment]
It's the intended behavior as it displays a realtime waveform of the current moment, not an aggregated waveform of the whole recorded audio.

**Reply by fatehux** [reply]
But the previously recorded sounds still vibrate it must be fixed not dancing up & down.

**Reply by katspaugh** [reply]
Could you record a video and post here? I haven't seen this vibration.

**Reply by fatehux** [reply]
Yes, sure

https://github.com/katspaugh/wavesurfer.js/assets/160303879/da947a7b-b1df-4dd6-8d0e-e40009ab60b4

**Reply by katspaugh** [reply]
I see, thanks! So it's like a subtle flickering on each frame while the waveform is being re-drawn. It looks like you're using wavesurfer v6 and the micrphone plugin?

**Reply by fatehux** [reply]
I am using   "wavesurfer.js": "^7.7.5",

**Reply by katspaugh** [reply]
The record plugin in v7 doesn’t have this mode that renders a waveform continuously, though. I guess you programmed that part yourself? The reason for flickering is probably that it recalculates the max value across a number of samples each time, so as new values come in, the max value changes ever so slightly.

**Reply by fatehux** [reply]
public handleRecord = () => {
    this.record = this.wavesurfer.registerPlugin(
      Record.create({
        scrollingWaveform: true,
        renderRecordedAudio: false,
        
      })
    );
  };
  this is my code i do not add anything.

**Reply by katspaugh** [reply]
Ah, true, the scrollingWaveform mode was added in https://github.com/katspaugh/wavesurfer.js/pull/3307, I forgot about that.
